addon-test-support/load.js

```diff
@@ -19,4 +22,69 @@ export default function loadEmberExam() {
   if (window.Testem) {
     patchTestemOutput(TestLoader);
   }
+
+  testLoader = new TestLoader();
 }
+
+// loadTests() is equivalent to ember-qunit's loadTest() except this does not create a new TestLoader instance
+function loadTests() {
+  testLoader.loadModules();
+}
+
+// setupQUnitCallBacks() registers QUnit callbacks neeeded for the load-balance option.
+function setupQUnitCallbacks(qunit) {
+  if (!location.search.includes("loadBalance")) {
+    return;
+  }
+  qunit.begin(function() {
+    return new Promise(function(resolve) {
+      Testem.on('testem:next-module-response', function getTestModule(moduleName) {
+        if (moduleName !== undefined) {
+          loadIndividualModule(moduleName);
+        }
+
+        // if no tests were added, request the next module
+        if (qunit.config.queue.length === 0) {
+          Testem.emit('get-next-module');
+        } else {
+          Testem.removeEventCallbacks('testem:next-module-response', getTestModule);
+          resolve();
+        }
+      });
+      Testem.emit('get-next-module');
+    });
+  });
+  qunit.moduleDone(function() {
+    return new Promise(function(resolve) {
+      Testem.on('testem:next-module-response', function getTestModule(moduleName) {
+        if (moduleName !== undefined) {
+          loadIndividualModule(moduleName);
+        }
+
+        // if no tests were added, request the next module
+        if (qunit.config.queue.length === 0) {
+          Testem.emit('get-next-module');
+        } else {
+          // `removeCallback` removes if the event queue contains the same callback for an event.
+          Testem.removeEventCallbacks('testem:next-module-response', getTestModule);
+          Testem.removeEventCallbacks('testem:module-queue-complete', resolve);
+          resolve();
+        }
+      });
+      Testem.on('testem:module-queue-complete', resolve);
+      Testem.emit('get-next-module');
+    });
+  });
+}
+
+// loadIndividualModule() executes loadIndividualModule() defined in patch-test-loader. It enables to load an AMD module one by one.
+function loadIndividualModule(moduleName) {
+  return testLoader.loadIndividualModule(moduleName);
```

Comment:
Does this function need to return at all? The return value is never used AFAICT.